### PR TITLE
use a PHP 8 compatible Composer version

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
using a version of Composer that includes composer/composer#9183 will make tests of the ErrorHandler component pass

File: .travis.yml [hunk 1]
```diff
@@ -196,6 +196,8 @@ before_install:
           ldapadd -h localhost:3389 -D cn=admin,dc=symfony,dc=com -w symfony -f src/Symfony/Component/Ldap/Tests/Fixtures/data/fixtures.ldif
       fi
 
+    - if [[ $PHP = nightly ]]; then composer self-update --snapshot --1; fi
```
Comment (stof) [inline .travis.yml:199]: passing both `--snapshot` and `--1` is useless. There is a single snapshot channel (and it is now about composer 2). `--1` means using the stable channel but restricted to `1.x` versions.

Comment (stof) [reply .travis.yml:199]: if you pass multiple options, the first one will win in the order specified in https://github.com/composer/composer/blob/c6d0309d48e57c8e9a46988bd71d34c4079256a6/src/Composer/SelfUpdate/Versions.php#L24

**Comment by stof** [comment]
Also, nothing actually works in the PHP nightly job, because composer complains about a missing `mongodb` extension.

File: .travis.yml (hunk 1 above)
Comment (stof) [reply .travis.yml:199]: see https://github.com/composer/composer/issues/9189 for a discussion on Composer about having a snapshot of 1.x

**Comment by stof** [comment]
> Also, nothing actually works in the PHP nightly job, because composer complains about a missing `mongodb` extension.

this might be related to using composer 2 rather than composer 1, but this might indicate a behavior difference.

**Pull request closed without merging**